internal/package_/goa.go

```diff
@@ -34,22 +35,31 @@ var patternMatchingCharReplacer = strings.NewReplacer(
 	"_", "\\_",
 )
 
-var ErrInvalidToken error = goapackage.Unauthorized("invalid token")
+var (
+	ErrUnauthorized error = goapackage.Unauthorized("Unauthorized")
+	ErrForbidden    error = goapackage.Forbidden("Forbidden")
+)
 
 func (w *goaWrapper) JWTAuth(
```

Comment:
Would it be useful to write a test for w.JWTAuth given its criticality? i.e. if we're not getting it right here then we're doomed. I realize we don't have yet a mock of TokenVerifier using GoMock but maybe one should be created, so we can do something like:

```go
// File: auth_test.go
package package_

func TestJWTAuth(t *testing.T) {
	w := &goaWrapper{
		packageImpl: &packageImpl{
			tokenVerifier: &auth.NoopTokenVerifier{},  // TODO: use GoMock.
		},
	}
	
	ctx, err := w.JWTAuth(context.Background(), "", &security.JWTScheme{})
	// ...
}
```

Reply:
Yeah, I was thinking about this yesterday, I was not planning to create a mock, but will do ;)

Reply:
I thought we had at least some tests for it, but it looks like we don't, so definitely better with a mock ;)

Reply:
This is WIP, I'll get back to it later today.